ZuluContent/Zulu/Spells/Earth/EarthsBlessingSpell.cs

```diff
@@ -1,86 +1,54 @@
 using System;
 using System.Collections;
+using System.Threading.Tasks;
 using Server.Network;
 using Server.Items;
 using Server.Mobiles;
 using Scripts.Zulu.Engines.Classes;
 using Server;
 using Server.Targeting;
 using Server.Spells;
+using ZuluContent.Zulu.Engines.Magic.Enchantments.Buffs;
 
 namespace Scripts.Zulu.Spells.Earth
 {
-    public class EarthsBlessingSpell : AbstractEarthSpell, IMobileTargeted
+    public class EarthsBlessingSpell : EarthSpell, IAsyncSpell
     {
-        public override TimeSpan CastDelayBase
-        {
-            get { return TimeSpan.FromSeconds(2); }
-        }
-
-        public override double RequiredSkill
-        {
-            get { return 60.0; }
-        }
-
-        public override int RequiredMana
-        {
-            get { return 10; }
-        }
+        public override TimeSpan CastDelayBase => TimeSpan.FromSeconds(0);
+    
+        public override double RequiredSkill => 60.0;
+    
+        public override int RequiredMana => 5;
 
         public EarthsBlessingSpell(Mobile caster, Item spellItem) : base(caster, spellItem)
         {
         }
 
-        public override void OnCast()
-        {
-            Caster.Target = new MobileTarget(this, 10, TargetFlags.Beneficial);
-        }
-
-        public void OnTargetFinished(Mobile from)
+        public async Task CastAsync()
         {
-            FinishSequence();
-        }
+            if (!Caster.CanBuff(Caster, true, BuffIcon.Curse, BuffIcon.Bless))
+                return;
 
-        public void OnTarget(Mobile from, Mobile m)
-        {
-            if (!Caster.CanSee(m))
+            var modAmount = (int) (SpellHelper.GetModAmount(Caster, Caster, StatType.All) * 1.5);
+            var duration = SpellHelper.GetDuration(Caster, Caster) * 1.5;
+            
+            Caster.TryAddBuff(new StatBuff(StatType.All)
```

Comment:
Should set the name and find a buff Icon the buffs here

Reply:
Ah yeah didn't think about that - I can just override the name with this { } syntax